tools/datagen/src/bin/verify-zero-copy.rs

```diff
@@ -0,0 +1,257 @@
+            keys.extend(paths.map(Cow::Borrowed));
+        }
+
+        let filtered: Vec<_> = all_keys
+            .into_iter()
+            .filter(|k| keys.contains(k.get_path()))
+            .collect();
+
+        if filtered.is_empty() {
+            eyre::bail!("No keys selected (or keys passed in do not exist), pass in --keys KEYNAME or --test-keys");
+        }
+
+        filtered
+    };
+
+    let cldr_json_root = if let Some(cldr) = matches.value_of("CLDR_ROOT") {
+        PathBuf::from(cldr)
+    } else if matches.is_present("INPUT_FROM_TESTDATA") {
+        icu_testdata::paths::cldr_json_root()
+    } else {
+        eyre::bail!("Value for --cldr--root must be specified (or --input-from-testdata)",)
+    };
+    let uprops_root = if let Some(uprops) = matches.value_of("UPROPS_PATH") {
+        PathBuf::from(uprops)
+    } else if matches.is_present("INPUT_FROM_TESTDATA") {
+        icu_testdata::paths::uprops_toml_root()
+    } else {
+        eyre::bail!("Value for --uprops-path must be specified (or --input-from-testdata)",)
+    };
+    let cldr_paths = Box::new(CldrPathsAllInOne {
+        cldr_json_root,
+        locale_subset: matches
+            .value_of("CLDR_LOCALE_SUBSET")
+            .unwrap_or("full")
+            .to_string(),
+    });
+
+    let converter: Box<dyn OmnibusDatagenProvider<SerializeMarker> + Sync> =
+        Box::new(ForkByKeyProvider(
+            icu_provider_cldr::create_exportable_provider(
+                cldr_paths.as_ref(),
+                uprops_root.clone(),
+            )?,
+            icu_provider_uprops::create_exportable_provider(&uprops_root)?,
+        ));
+
+    let selected_locales = icu_testdata::metadata::load()?.package_metadata.locales;
+
+    let converter = Box::new(
+        converter
+            .filterable("icu4x-datagen locales")
+            .filter_by_langid_allowlist_strict(&selected_locales),
+    );
+
+    let postcard_file = if let Some(file) = matches.value_of("POSTCARD_FILE") {
+        PathBuf::from(file)
+    } else if matches.is_present("INPUT_FROM_TESTDATA") {
+        icu_testdata::paths::data_root().join("testdata.postcard")
+    } else {
+        eyre::bail!("Value for --postcard-file must be specified (or --input-from-testdata)",)
+    };
+    let mut blob: Vec<u8> = Vec::new();
+    File::open(postcard_file)
+        .expect("File should exist")
+        .read_to_end(&mut blob)
+        .expect("Reading pre-computed postcard buffer");
+    // Create a DataProvider from it:
+    let provider =
+        BlobDataProvider::new_from_rc_blob(Rc::from(blob)).expect("Deserialization should succeed");
+
+    // Litemap keeps it sorted, convenient
+    let mut violations: LiteMap<&'static str, u64> = LiteMap::new();
+
+    for key in selected_keys.into_iter() {
+        let props_key = key.get_path().starts_with("props/");
+
+        let mut max_violation = 0;
+
+        for options in converter.supported_options_for_key(key)? {
```

Comment:
Suggestion: Actually it is kind-of odd to use the supported_options from the source data and then load it from the postcard file. I'm warming up to the idea that we should read from source data, serialize, and deserialize. This can be in another PR.

Reply:
The idea was that (a) the blob provider doesn't know about supported options and (b) i'd like to check for all options that are _supposed_ to be in the file.